src/streamlink/plugins/crunchyroll.py

```diff
@@ -352,14 +372,12 @@ def _get_streams(self):
         return streams
 
     def _create_api(self):
-        """Creates a new CrunchyrollAPI object, initiates it's session and
+        """Creates a new CrunchyrollAPI object, initiates its session and
         tries to authenticate it either by using saved credentials or the
         user's username and password.
         """
         if self.options.get("purge_credentials"):
-            self.cache.set("session_id", None, 0)
-            self.cache.set("auth", None, 0)
-            self.cache.set("session_id", None, 0)
+            self.cache.set("auth", None, expires=0)
```

Comment:
I'm not sure why `self.cache.set("session_id", None, 0)` was called twice here.  I'm not sure why it's called at all, as there is nothing that sets the `session_id` in the cache.

I wonder if it is worth also scrubbing the `device_id` from the cache when the credentials are purged?

Reply:
> was called twice here

https://github.com/streamlink/streamlink/blame/bcd624c1753404e7119837a4e050f785485818d3/src/streamlink/plugins/crunchyroll.py#L360-L362
Probably a mistake in 1ced3256ae257ae7bb19675a6d68ac24aa4e7a6c

> also scrubbing the `device_id` from the cache

Purge credentials should clear the entire cache.